packages/toolkit/src/repository/repository.ts

```diff
@@ -556,33 +563,42 @@ export class DappnodeRepository extends ApmRepository {
    * @param format - The format to parse the data into.
    * @returns The parsed data.
    */
-  private parseAsset<T>(data: string, format: FileFormat): T {
-    switch (format) {
-      case FileFormat.YAML:
-        try {
-          const parsedData = YAML.parse(data);
-          if (!parsedData || typeof parsedData === "string")
-            throw Error(`returned invalid object`);
-          return parsedData as T;
-        } catch (e) {
-          if (e instanceof Error)
-            e.message = `Error parsing YAML: ${e.message}`;
-          throw e;
+  private parseAsset<T>(data: string | string[], format: FileFormat): T {
+    const parseSingle = (content: string): any => {
+      switch (format) {
+        case FileFormat.YAML: {
+            const parsedYaml = YAML.parse(content);
+            if (!parsedYaml || typeof parsedYaml === "string")
+              throw new Error("Invalid YAML object");
+            return parsedYaml;
         }
-      case FileFormat.JSON:
-        try {
-          return JSON.parse(data);
-        } catch (e) {
-          if (e instanceof Error)
-            e.message = `Error parsing JSON: ${e.message}`;
-          throw e;
+        case FileFormat.JSON: {
+            return JSON.parse(content);
         }
-      case FileFormat.TEXT:
-        return data as T;
-      default:
-        throw Error(`Attempting to parse unknown format ${format}`);
+        case FileFormat.TEXT: {
+          return content; // TEXT format assumes direct usage of the string.
+        }
+        default: {
+          throw new Error(`Unsupported format: ${format}`);
+        }
+      }
+    };
+  
+    try {
+      if (Array.isArray(data)) {
+        // Map over array data if it's an array, using parseSingle for each item
+        return data.map(parseSingle) as unknown as T;
```

Comment:
Using `as unknown` is considered a really bad practice in ts. Is there any other workaround in this approach?

Reply:
fixed, removed "unknown" and "any" types.